src/fit_to_points/get_control_points.rs

```diff
@@ -0,0 +1,87 @@
+use glifparser::{Contour, Point, PointType};
+// https://www.codeproject.com/Articles/31859/Draw-a-Smooth-Curve-through-a-Set-of-2D-Points-wit
+
+pub fn get_curve_control_point(knots: Contour<()>) -> (Contour<()>, Contour<()>) {
+    let n = knots.len() - 1;
+
+    let mut first_control_point = Contour::new();
+    let mut second_control_point = Contour::new();
+    if n == 1 {
+        // Special case: Bezier curve should be a straight line.
+        // 3P1 = 2P0 + P3
+
+        first_control_point.push(Point::from_x_y_type(
+            (
+                (2. * knots[0].x + knots[1].x) / 3.,
+                (2. * knots[0].y + knots[1].y) / 3.,
+            ),
+            PointType::Undefined,
```

Comment:
Using this point type is dangerous. It only exists for use in `GlifPoint` and must not be used in `Point`'s.

Reply:
So can I use a `Vector` here

Reply:
Yes.

Reply:
Or actually `PointType::Move` is OK too if they are both single floating points.

Reply:
There's also `PointType::OffCurve`, which is quite dangerous to be put in a `Contour`, but if you go over it carefully and make sure no `PointType::OffCurve` in final `Contour` it's okay.

Reply:
can I use `PointType::Move` for every point.

Reply:
Basically, `Point` should never be `Undefined`, and `Contour` should only ever contain `Move`, `Line`, or `Curve` _at the time API returns_. You can use the other types while processing, that's what they're for. `Undefined` is only for `GlifPoint`.

Reply:
I actually don't need a `Point` type over there I just have to return two `f32` ,So I will use `Vector` or `(f32,f32)`

Reply:
Yes, good idea.

Reply:
I tried to use `Vector` but there the types are in `f64` so then I used `(f32,f32)`